src/Features/LanguageServer/ProtocolUnitTests/ProjectContext/GetTextDocumentWithContextHandlerTests.cs

```diff
@@ -0,0 +1,109 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable enable
+
+using System;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Roslyn.Test.Utilities;
+using Xunit;
+using LSP = Microsoft.VisualStudio.LanguageServer.Protocol;
+
+namespace Microsoft.CodeAnalysis.LanguageServer.UnitTests.ProjectContext
+{
+    public class GetTextDocumentWithContextHandlerTests : AbstractLanguageServerProtocolTests
+    {
+        [Fact]
+        public async Task SingleDocumentReturnsSingleContext()
+        {
+            var workspaceXml =
+@"<Workspace>
+    <Project Language=""C#"" CommonReferences=""true"" AssemblyName=""CSProj"">
+        <Document FilePath = ""C:\C.cs"">{|caret:|}</Document>
+    </Project>
+</Workspace>";
+
+            using var workspace = CreateXmlTestWorkspace(workspaceXml, out var locations);
+            var documentUri = locations["caret"].Single().Uri;
+            var result = await RunGetProjectContext(workspace.CurrentSolution, documentUri);
+
+            Assert.NotNull(result);
+            Assert.Equal(0, result!.DefaultIndex);
+            var context = Assert.Single(result.ProjectContexts);
+
+            Assert.Equal(ProtocolConversions.ProjectIdToProjectContextId(workspace.CurrentSolution.ProjectIds.Single()), context.Id);
+            Assert.Equal(LSP.ProjectContextKind.CSharp, context.Kind);
+            Assert.Equal("CSProj", context.Label);
+        }
+
+        [Fact]
+        public async Task MultipleDocumentsReturnsMultipleContexts()
+        {
+            var workspaceXml =
+@"<Workspace>
+    <Project Language=""C#"" CommonReferences=""true"" AssemblyName=""CSProj1"">
+        <Document FilePath=""C:\C.cs"">{|caret:|}</Document>
+    </Project>
+    <Project Language=""C#"" CommonReferences=""true"" AssemblyName=""CSProj2"">
+        <Document IsLinkFile=""true"" LinkFilePath=""C:\C.cs"" LinkAssemblyName=""CSProj1"">{|caret:|}</Document>
+    </Project>
+</Workspace>";
+
+            using var workspace = CreateXmlTestWorkspace(workspaceXml, out var locations);
+            var documentUri = locations["caret"].Single().Uri;
+            var result = await RunGetProjectContext(workspace.CurrentSolution, documentUri);
+
+            Assert.NotNull(result);
+
+            Assert.Collection(result!.ProjectContexts.OrderBy(c => c.Label),
```

Comment:
does Assert.NotNull not count as a null check?

Reply:
We're not on a version new enough to do that, and there's a bunch of TFM complexity, sadly.